app/controllers/SubmitArrangementController.scala

```diff
@@ -48,16 +52,26 @@ class SubmitArrangementController @Inject() (
     }
   }
 
-  private def submitArrangementAndUpdateJourney(journey: Journey.Stages.AgreedTermsAndConditions)(implicit request: Request[_]): Future[Result] = {
-    for {
-      arrangementResponse <- ttpService
-        .submitArrangement(journey)
-      _ <- journeyService
-        .updateArrangementResponse(journey.id, arrangementResponse)
-        .recover {
-          case e: Exception => Errors.throwServerErrorException(e.getMessage)
-        }
-    } yield Redirect(routes.PaymentPlanSetUpController.paymentPlanSetUp())
+  private def submitArrangementAndUpdateJourney(
+      journey: Journey.Stages.AgreedTermsAndConditions
+  )(implicit request: AuthenticatedJourneyRequest[_]): Future[Result] = {
+    ttpService.submitArrangement(journey)
+      .map { (r: ArrangementResponse) =>
+        auditService.auditPaymentPlanSetUp(journey, Right(r))
+        r
+      }.recover {
+        case aHttpError: HttpException =>
```

Comment:
I'm struggling to get past this name `aHttpError` 🙂

Reply:
haha, I'll update